lib/src/authentication/credential.dart

```diff
@@ -44,14 +54,19 @@ class Credential {
 
     final hostedUrl = validateAndNormalizeHostedUrl(json['url'] as String);
 
-    const knownKeys = {'url', 'token'};
+    const knownKeys = {'url', 'token', 'env'};
     final unknownFields = Map.fromEntries(
         json.entries.where((kv) => !knownKeys.contains(kv.key)));
 
+    String _optional(String key) {
+      return json[key] is String ? json[key] as String : null;
```

Comment:
I **think** this can be implemented as just: 
```suggestion
      return json[key] as String;
```

Reply:
This might throw an error if field is in another type (int, map, list). I think, I should throw FormatException instead to make sure field type is String or null.

Reply:
Yes - of course - I looked at this backwards...